### PR TITLE
Removing explicit LiquibaseServletListener references

### PR DESCRIPTION
**Pull request by Thunderforge**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

`GenericServletListener` can be either a `LiquibaseServletListener` or a `LiquibaseJakartaServletListener`.

Removing all explicit references to the former so as not to provide confusion when `LiquibaseJakartaServletListener` is used instead.

## Things to be aware of

None

## Things to worry about

None

## Additional Context

None

### CODE REVIEW

**Comment by nvoxland** [comment]
**Initial/Pre-Review Thoughts**

Thanks for the PR, @Thunderforge !

The changes make sense to me, it's cleaning up the docs and an error message which is always good.

Questions I have:
- None

Potential risks:
- None, it updates docs and a log message. No functionality changes

What could make the full review difficult:
- Nothing

**Comment by Thunderforge** [comment]
@MalloD12 It looks like you attempted a rebase, but it didn't go right. Now this PR has an additional 67 commits and affects over 200 files.

How should we proceed?

**Comment by filipelautert** [comment]
Hey @Thunderforge @MalloD12 - did you have only 1 commit to this branch being it https://github.com/liquibase/liquibase/pull/3677/commits/5f637ac79af07648af7ecbcaaf73c9ff92b9b440 ? 
You can pull it and force-push to this branch to fix the problematic merge. Then rebase again :crossed_fingers:

**Comment by MalloD12** [comment]
> @MalloD12 It looks like you attempted a rebase, but it didn't go right. Now this PR has an additional 67 commits and affects over 200 files.
> 
> How should we proceed?

Hi @Thunderforge,

Those additional commits were not because of the rebase. Well, they are in some way, but this is a GH cache issue it appears sometimes and you can simply fix it by changing the destiny branch. So, if you see it again you will see it only has two commits. 

Thanks,
Daniel.

**Review by MalloD12 — APPROVED** [review]
**Approved**.

## Review and testing results:

Changes made on this PR lgtm. Thank you @Thunderforge!

### Things to be aware of:
- None

### Things to worry about:
- None

**Review by suryaaki2 — APPROVED** [review]

**Review by XDelphiGrl — APPROVED** [review]
A `GenericServletListener` can be either 
a  `LiquibaseServletListener` or a `LiquibaseJakartaServletListener`, but outdated Java documentation in the `GenericServletListener` class indicated only `LiquibaseServletListener`. The confusion was further compounded by a hard-coded string, "LiquibaseServletListener," in the `checkPreconditions` method. This PR updates the Java docs and uses the class shortname to populate the error message with the correct listener type.
* Functional and test harness executions passing.
* No additional testing necessary.

APPROVED

**Comment by Thunderforge** [comment]
It looks like there are three approvals, but it still says "At least 1 approving review is required by reviewers with write access."

**Review by filipelautert — APPROVED** [review]

**Review by filipelautert — APPROVED** [review]
@Thunderforge  It's waiting for me to press the merge button ;)
Thanks for the code!

**Pull request merged**
